packages/effect/src/Schema.ts

```diff
@@ -4557,6 +4557,40 @@ export class ULID extends String$.pipe(
   })
 ) {}
 
+/**
+ * @category URL constructors
+ * @since 3.11.0
+ */
+export class URLFromSelf extends instanceOf(URL, {
+  identifier: "URLFromSelf",
+  title: "URLFromSelf",
+  arbitrary: (): LazyArbitrary<URL> => (fc) => fc.webUrl().map((s) => new URL(s)),
+  pretty: () => (url) => url.toString()
+}) {}
+
+/** @ignore */
+export class URL$ extends transformOrFail(String$, URLFromSelf, {
```

Comment:
I wouldn't export the class here.

I'd also add an annotation to `String$` as we do elsewhere, like

```ts
String$.annotations({ description: "a string that will be parsed into a URL" })
```

(useful when generating JSON Schemas)